### ISSUE TITLE
DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.

### ISSUE THREAD

**Issue by jakul** [pallets/jinja#942]
Setuptools [40.8.0](https://setuptools.readthedocs.io/en/latest/history.html#v40-8-0) has added a new warning which Jinja's PackageLoader is triggering:

DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.

I think this is caused by the leading slash added by this code:
```
> /Users/craig/src/boughtbymany/servicekit/.tox/py36-flask/lib/python3.6/site-packages/jinja2/loaders.py(235)get_source()
    234         p = '/'.join((self.package_path,) + tuple(pieces))
```

---

## Expected Behavior
No warnings.

## Actual Behavior
Warning is seen.

## Steps to reproduce
```
$ cd /tmp

[py36-flask] craig@amogato3:/tmp
$ mkdir res/templates -p
mkdir: res: No such file or directory

[py36-flask] craig@amogato3:/tmp
$ mkdir res

[py36-flask] craig@amogato3:/tmp
$ mkdir res/templates

[py36-flask] craig@amogato3:/tmp
$ touch res/templates/template.html

[py36-flask] craig@amogato3:/tmp
$ python -Wd
Python 3.6.7 (default, Dec  3 2018, 11:14:05)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import setuptools
/Users/craig/src/boughtbymany/servicekit/.tox/py36-flask/lib/python3.6/distutils/__init__.py:4: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
>>> setuptools.__version__
'40.8.0'

[py36-flask] craig@amogato3:/tmp
$ touch res/__init__.py

[py36-flask] craig@amogato3:/tmp
$ python -Wd
Python 3.6.7 (default, Dec  3 2018, 11:14:05)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from jinja2 import *
>>> Environment(loader=PackageLoader('res', '')).get_template('templates/template.html')
/Users/craig/src/boughtbymany/servicekit/.tox/py36-flask/lib/python3.6/site-packages/jinja2/loaders.py:235: DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.
  if not self.provider.has_resource(p):
/Users/craig/src/boughtbymany/servicekit/.tox/py36-flask/lib/python3.6/site-packages/jinja2/loaders.py:240: DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.
  filename = self.provider.get_resource_filename(self.manager, p)
/Users/craig/src/boughtbymany/servicekit/.tox/py36-flask/lib/python3.6/site-packages/jinja2/loaders.py:248: DeprecationWarning: Use of .. or absolute path in a resource path is not allowed and will raise exceptions in a future release.
  source = self.provider.get_resource_string(self.manager, p)
<Template 'templates/template.html'>
>>>

```

## Your Environment
* Python version: 3.6.7
* Jinja version: 2.10

**Comment by davidism** [comment]
I do not use the package loader, so I'm not very familiar with the setuptools resource API. What should be used instead? I'm happy to review a PR.

**Comment by jakul** [comment]
I think it's just a case of replacing `p = '/'.join((self.package_path,) + tuple(pieces))` with `p = '/'.join((self.package_path,) + tuple(pieces))[1:]` so that the output doesn't start with a slash

**Comment by jakul** [comment]
I'm getting a leading `/` because the `package_path` argument I send is `''`. This is because my templates are not located under any particular directory and are spread across my entire codebase, so there's no single "top level" prefix I can give here

**Comment by davidism** [comment]
Closed by #1082
